primitives/state-machine/src/overlayed_changes/mod.rs

```diff
@@ -90,6 +90,23 @@ pub struct StorageChanges<Transaction, H: Hasher, N: BlockNumber> {
 	pub changes_trie_transaction: Option<ChangesTrieTransaction<H, N>>,
 }
 
+impl<Transaction, H: Hasher, N: BlockNumber> Clone for StorageChanges<Transaction, H, N>
+where
+    H::Out: Clone,
+    Transaction: Clone,
+{
+    fn clone(&self) -> Self {
+        StorageChanges {
+            main_storage_changes: self.main_storage_changes.clone(),
+            child_storage_changes: self.child_storage_changes.clone(),
+            offchain_storage_changes: self.offchain_storage_changes.clone(),
+            transaction: self.transaction.clone(),
+            transaction_storage_root: self.transaction_storage_root.clone(),
+            changes_trie_transaction: None,
```

Comment:
I don't exactly remember how this field is used but what is the reason not to clone it ?

Reply:
Well, that's a good question and I'm also not certain about its purpose. Type `ChangesTrieTransaction<H, N>` contains `MemoryDB<H>` and I just assumed we don't want to clone it around. My assumption is that other fields provide IDs to the same database where data still exists but I may be wrong.

Reply:
I'm using a `StorageChanges` instance from block proposer and it's always `None` anyway.

Reply:
I think either we do:
* add function: `fn clone_no_changes_trie_transaction`
* or we also clone the change_trie_transaction.

But otherwise it seems quite unexpected behavior to me. (Though I'm not very familiar with this part of the code)

Reply:
Is `clone_no_changes_trie` OK?

Reply:
Yes, please clone them as well.